py/objgenerator.c

```diff
@@ -313,11 +319,14 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_1(gen_instance_close_obj, gen_instance_close);
 #if MICROPY_PY_GENERATOR_PEND_THROW
 STATIC mp_obj_t gen_instance_pend_throw(mp_obj_t self_in, mp_obj_t exc_in) {
     mp_obj_gen_instance_t *self = MP_OBJ_TO_PTR(self_in);
-    if (self->code_state.sp == self->code_state.state - 1) {
-        mp_raise_TypeError("can't pend throw to just-started generator");
+    if (exc_in == mp_const_none) {
+        mp_raise_ValueError(NULL);
+    }
```

Comment:
Why is this restriction needed?  Can this check just be removed?  If the user passes in `None` then it just cancels any pending throw, which seems sensible/possible...?

Reply:
Yes, you're right. Removed and updated the test to check that this works.

I think this was left over from when I had the meaning of `MP_OBJ_NULL` and `mp_const_none` the other way around.